src/common/log.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import log, {LevelOption} from 'electron-log';
+
+import Util from 'common/utils/util';
+
+export const setLoggingLevel = (level: string) => {
+    if (log.transports.file.level === level) {
+        return;
+    }
+    log.error('Logger', 'Log level set to:', level);
+
+    log.transports.console.level = level as LevelOption;
+    log.transports.file.level = level as LevelOption;
+};
+
+export class Logger {
+    private prefixes: string[];
+
+    constructor(...prefixes: string[]) {
+        this.prefixes = prefixes;
+    }
+
+    withPrefix = (...prefixes: string[]) => {
+        return {
+            error: this.prefixed(log.error, ...prefixes),
+            warn: this.prefixed(log.warn, ...prefixes),
+            info: this.prefixed(log.info, ...prefixes),
+            verbose: this.prefixed(log.verbose, ...prefixes),
+            debug: this.prefixed(log.debug, ...prefixes),
+            silly: this.prefixed(log.silly, ...prefixes),
+            log: this.prefixed(log.log, ...prefixes),
+        };
+    }
+
+    private prefixed = (func: (...args: any[]) => void, ...additionalPrefixes: string[]) => {
+        return (...args: any[]) => func(...this.shortenPrefixes(...this.prefixes, ...additionalPrefixes), ...args);
+    }
+
+    private shortenPrefixes = (...prefixes: string[]) => {
+        return prefixes.map((prefix) => `[${Util.shorten(prefix)}]`);
+    }
+
+    error = this.prefixed(log.error);
+    warn = this.prefixed(log.warn);
+    info = this.prefixed(log.info);
+    verbose = this.prefixed(log.verbose);
+    debug = this.prefixed(log.debug);
+    silly = this.prefixed(log.silly);
+    log = this.prefixed(log.log);
```

Comment:
If I am reading this right, any change on `this.prefixes` won't be visible here, since `this.prefixed` is executed on object creation. If we wanted to make this work as is (that I don't recommend) we would need something like:
`error = (args) => this.prefixed(log.error)(args)`

If we don't want the prefixes to change after logger creation, we can always define all this on the object constructor, and not store the prefixes on `this.prefixes`.

Reply:
I would assume `this.prefixes` is immutable, hence why it's private. Unfortunately JavaScript doesn't have an `ImmutableSet` so I think the array works fine.